### PR TITLE
Add parameterless overloads of TrimStart/TrimEnd

### PR DESCRIPTION
**Pull request by thiagocamargos**
Add parameterless overloads of TrimStart/TrimEnd on String.Manipulation.cs, model.xml and ref/mscorlib.cs

Fix https://github.com/dotnet/corefx/issues/6485

### CODE REVIEW

**Comment by thiagocamargos** [comment]
fix #corefx/6485

**Comment by fujiy** [comment]
Fix #dotnet/corefx/6485

File: src/mscorlib/src/System/String.Manipulation.cs [hunk 1]
```diff
@@ -1553,15 +1553,24 @@ private String[] SplitInternal(String separator, String[] separators, Int32 coun
             }
             return TrimHelper(trimChars,TrimHead);
         }
-    
-    
+
+        // Removes a set of characters from the beginning of this string.
+        public String TrimStart() {
+            return TrimHelper(TrimHead);
```
Comment (joperezr) [inline src/mscorlib/src/System/String.Manipulation.cs]: Nit: You could simplify this by simply doing: 
```c#
public string TrimStart() => TrimHelper(TrimHead);
public string TrimEnd() => TrimHelper(TrimTail);
```

**Comment by thiagocamargos** [comment]
Hi @joperezr 
I see there are already some tests on https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/StringTests.cs#L2346 and https://github.com/dotnet/corefx/blob/master/src/System.Runtime/tests/System/StringTests.cs#L2328
that used the TrimStart/End(params char []) overload and will use now the parameterless overload.

Should I adapt the test to ensure it will also keep calling the params char [] overload when trimChars is null?

**Comment by joperezr** [comment]
> Should I adapt the test to ensure it will also keep calling the params char [] overload when trimChars is null

Yes, I believe you should ensure it keeps calling the other params one so that we don't loose coverage there and perhaps add some basic ones for this two methods as well.

**Comment by joperezr** [comment]
@thiagocamargos thanks for your contribution!

**Comment by jkotas** [comment]
@thiagocamargos Would you mind porting this change to CoreRT (https://github.com/dotnet/corert/tree/master/src/System.Private.CoreLib/src) as well?

**Comment by thiagocamargos** [comment]
Hi @jkotas 
Sure, I'll do that.

**Comment by justinvp** [comment]
@thiagocamargos, I included these in https://github.com/dotnet/corert/pull/2587 while I was porting my changes from #9009. Sorry if you were already working on it.

**Comment by thiagocamargos** [comment]
@justinvp no problem, I could not work on last couple weeks

**Pull request merged**